e3nn/o3/_tensor_product/_sub.py

```diff
@@ -205,6 +205,72 @@ def __init__(
         super().__init__(irreps_in1, irreps_in2, out, instr, irrep_normalization=irrep_normalization, **kwargs)
 
 
+class FullTensorProductSHWeighted(TensorProduct):
+    r"""Nequip-like TensorProduct with weights.
+
+    .. math::
+
+        `z_u = x_u \otimes (\sum_v w_{uv} y_v)`
```

Comment:
This should be z_w, not z_u (since the output irreps are more than the input irreps)